discord/presences.py

```diff
@@ -0,0 +1,166 @@
+
+    @classmethod
+    def _copy(cls, client_status: Self, /) -> Self:
+        self = cls.__new__(cls)  # bypass __init__
+
+        self._status = client_status._status
+
+        self.desktop = client_status.desktop
+        self.mobile = client_status.mobile
+        self.web = client_status.web
+
+        return self
+
+    @property
+    def status(self) -> Status:
+        """:class:`Status`: The user's overall status. If the value is unknown, then it will be a :class:`str` instead."""
+        return try_enum(Status, self._status)
+
+    @property
+    def raw_status(self) -> str:
+        """:class:`str`: The user's overall status as a string value."""
+        return self._status
+
+    @property
+    def mobile_status(self) -> Status:
+        """:class:`Status`: The user's status on a mobile device, if applicable."""
+        return try_enum(Status, self.mobile or 'offline')
+
+    @property
+    def desktop_status(self) -> Status:
+        """:class:`Status`: The user's status on the desktop client, if applicable."""
+        return try_enum(Status, self.desktop or 'offline')
+
+    @property
+    def web_status(self) -> Status:
+        """:class:`Status`: The user's status on the web client, if applicable."""
+        return try_enum(Status, self.web or 'offline')
+
+    def is_on_mobile(self) -> bool:
+        """A helper function that determines if a user is active on a mobile device.
+
+        Returns
+        -------
+        :class:`bool`
+        """
+        return self.mobile is not None
+
+
+class RawPresenceUpdateEvent(_RawReprMixin):
+    """Represents the payload for a :func:`on_raw_presence_update` event.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    user_id: :class:`int`
+        The ID of the user that triggered the presence update.
+    guild_id: Optional[:class:`int`]
+        The guild ID for the users presence update. Could be ``None``.
+    guild: Optional[:class:`Guild`]
+        The guild associated with the presence update and user. Could be ``None``.
+    client_status: :class:`ClientStatus`
+        The :class:`~.ClientStatus` model which holds information about the status of the user on various clients.
+    """
+
+    __slots__ = ('user_id', 'guild_id', 'guild', 'client_status', '_activities')
+
+    def __init__(self, *, data: PartialPresenceUpdate, state: ConnectionState) -> None:
+        self.user_id: int = int(data['user']['id'])
+
+        self.client_status: ClientStatus = ClientStatus(status=data['status'], data=data['client_status'])
+        self._activities: Union[Tuple[ActivityTypes, ...], None] = None
+
+        self.guild_id: Optional[int] = _get_as_snowflake(data, 'guild_id')
+        self.guild: Union[Guild, None] = state._get_guild(self.guild_id)
+
+    def _create_activities(self, data: PartialPresenceUpdate, state: ConnectionState) -> None:
+        self._activities = tuple(create_activity(d, state) for d in data['activities'])
```

Comment:
I don't think this should be done outside of `__init__`, as it'll be called on every `RawPresenceUpdate` anyway since you either dispatch `raw_presence_update`, call `Member._presence_update`, or even do both.